File: ps/hamming-distance.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        int res=0,k;
        k=x^y;
        while(k!=0){
            if (k%2==1) res++;
            k >>= 1;
        }
        return res;
    }
};

File: ps/hamming-distance.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        res = 0
        while x > 0 or y > 0:
            if x % 2 != y % 2:
                res += 1
            x //= 2
            y //= 2
        return res